.github/workflows/pr.yml

```diff
@@ -9,7 +9,7 @@ jobs:
     name: shared-operator-workflow
     uses: redhat-cop/github-workflows-operators/.github/workflows/pr-operator.yml@v1.0.8
     with:
-      GO_VERSION: ~1.21
+      GO_VERSION: ~1.23
```

Comment:
The current version of `controller-runtime` is not compatible with golang 1.23. Would recommend omitting the change of go version from this PR so that it can be handled separately as part of a larger set of dependency updates

Reply:
@sabre1041 I have restored the version to 1.21. I noticed that there was a mismatch between the version declared in the workflow and go.mod, and with the version of the base image. I have kept the change in the base image to match 1.21. Please let me know if this also needs to go back to 1.18 for some reason.